<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="ScreenR">
<title>ScreenR Example Analysis • ScreenR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ScreenR Example Analysis">
<meta property="og:description" content="ScreenR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ScreenR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.41</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Analysis_Example.html">ScreenR Example Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EmanuelSoda/ScreenR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ScreenR Example Analysis</h1>
                        <h4 data-toc-skip class="author">Emanuel Michele
Soda</h4>
            <address class="author_afil">
      Istituto Europeo Oncologia (IEO), Milano,
Italy<br><a class="author_email" href="mailto:#"></a><a href="mailto:emanuelmichele@ieo.it" class="email">emanuelmichele@ieo.it</a> or <a href="mailto:emanuelsoda@gmail.com" class="email">emanuelsoda@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/EmanuelSoda/ScreenR/blob/HEAD/vignettes/Analysis_Example.Rmd" class="external-link"><code>vignettes/Analysis_Example.Rmd</code></a></small>
      <div class="d-none name"><code>Analysis_Example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="importing-pacakge">Importing Pacakge<a class="anchor" aria-label="anchor" href="#importing-pacakge"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>fig.width<span class="op">=</span><span class="fl">10</span>, fig.height<span class="op">=</span><span class="fl">7</span><span class="op">)</span> </code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The R package <strong>ScreenR</strong> has been developed to perform
the analysis of data coming from RNA-seq data generated using Genetic
Screening. It is based on the same idea of edgeR but it also integrate
the idea at the base of Tidyverse.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p><strong>ScreenR</strong> requires several CRAN and Bioconductor R
packages to be installed. Dependencies are usually handled
automatically, when installing the package using the following
commands:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ScreenR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="manual-installation">Manual installation<a class="anchor" aria-label="anchor" href="#manual-installation"></a>
</h3>
<p>In the unlikely case that a manual installation is required, e.g., if
you do not install <strong>ScreenR</strong> via Bioconductor (which is
highly recommended), you can install CRAN pacakges using:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"&lt;package_name&gt;"</span><span class="op">)</span></code></pre></div>
<p>While Bioconductor packages can be installed from R using the
following command:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"&lt;package_name&gt;"</span><span class="op">)</span></code></pre></div>
<p>Sometimes, it may also be useful to update Bioconductor:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Finally, the manual installation of <strong>ScreenR</strong> can, for
example, be done from the command line …</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>R CMD INSTALL ScreenR_<span class="sc">&lt;</span>version<span class="sc">&gt;</span>.tar.gz</span></code></pre></div>
<p>… or the newest version can directly be installed from GitHub using
the CRAN package devtools:</p>
</div>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<div class="section level3">
<h3 id="loading-the-package">Loading the package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<p>After installation, loading the package is simple as :</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://emanuelsoda.github.io/ScreenR/" class="external-link">ScreenR</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="read-data">Read Data<a class="anchor" aria-label="anchor" href="#read-data"></a>
</h3>
<p>The input of ScreenR is a count table. A count table is usually the
starting point of an RNA-seq differentially expressed genes analysis and
consists of a matrix containing reads count organized with:</p>
<ul>
<li>Genes on the rows</li>
<li>Samples on the columns</li>
</ul>
<p>For this vignette we will use as an example a Loss of Function
Genetic Screening performed using . First of all the data has to be
read.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">count_table</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">annotation_table</span><span class="op">)</span>

<span class="va">data</span> <span class="op">&lt;-</span> <span class="va">count_table</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"Barcode"</span>, <span class="st">"T1"</span>, <span class="st">"T2"</span>, <span class="st">"Time3_TRT_A"</span>, <span class="st">"Time3_TRT_B"</span>, <span class="st">"Time3_TRT_C"</span>,
    <span class="st">"Time3_A"</span>, <span class="st">"Time3_B"</span>, <span class="st">"Time3_C"</span>, <span class="st">"Time4_TRT_A"</span>, <span class="st">"Time4_TRT_B"</span>,
    <span class="st">"Time4_TRT_C"</span>, <span class="st">"Time4_A"</span>, <span class="st">"Time4_B"</span>, <span class="st">"Time4_c"</span>
<span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">Barcode</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Barcode</span> <span class="op">!=</span> <span class="st">"*"</span><span class="op">)</span>


<span class="va">total_Annotation</span> <span class="op">&lt;-</span>  <span class="va">annotation_table</span> </code></pre></div>
</div>
<div class="section level3">
<h3 id="object-creation">Object Creation<a class="anchor" aria-label="anchor" href="#object-creation"></a>
</h3>
<p>The second needed step is to create a <strong>ScreenR object</strong>
from the count table. The ScreenR object is created using the function
<strong>create_screenR_object()</strong>. This object will be used to
store the most important information to perform the analysis. Most of
the ScreenR function takes as main input the ScreenR object to perform
the needed operation and return a result.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)_\\w+"</span>, <span class="st">"\\1"</span>, <span class="va">groups</span><span class="op">)</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">groups</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span><span class="op">)</span>

<span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#8da0cb"</span>, <span class="fl">3</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#e78ac3"</span>, <span class="fl">3</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#a6d854"</span>, <span class="fl">3</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#ffd92f"</span>, <span class="fl">3</span><span class="op">)</span>
<span class="op">)</span>

<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_screenR_object.html">create_screenR_object</a></span><span class="op">(</span>
    table <span class="op">=</span> <span class="va">data</span>, annotation <span class="op">=</span> <span class="va">total_Annotation</span>, groups <span class="op">=</span> <span class="va">groups</span>,
    replicates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="computing-the-needed-tables">Computing the needed tables<a class="anchor" aria-label="anchor" href="#computing-the-needed-tables"></a>
</h3>
<p>In order to start the analysis the data has first to be normalized
and the <em>data_table</em>.</p>
<p>For what concern the normalizzation ScreenR uses a normalizzation
that is called <em>Reads Per Million Sample (RPMS)</em> which has the
following mathematical expression:</p>
<p><span class="math display">\[RPMS = \frac{Number \; of \; mapped \;
reads \; to \; a \; barcode}
                   { \sum_{sample}{Number\; of \;mapped \; reads}}
*10^{6}\]</span></p>
<p>The number of reads mapped for each Barcode in a sample are
normalized by the number of reads in that sample and multiplied by one
million. Then has to be computed the <em>data_table</em>. The
<em>data_table</em> can be seen as a tidy version of the original
<em>count table</em> and will be used throughout the analysis.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_data.html">normalize_data</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_data_table.html">compute_data_table</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mapped-reads">Mapped Reads<a class="anchor" aria-label="anchor" href="#mapped-reads"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_mapped_reads.html">plot_mapped_reads</a></span><span class="op">(</span><span class="va">object</span>, <span class="va">palette</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/comma.html" class="external-link">number_format</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Number of Mapped Reads in each sample"</span><span class="op">)</span>

<span class="va">plot</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_mapped_reads-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="quality-check">Quality Check<a class="anchor" aria-label="anchor" href="#quality-check"></a>
</h3>
<p>The first step to perform when dealing with RNA-seq data is to check
the quality of the samples. In ScreenR this can be done using several
methods. For example the distribution can be seen using both boxplots or
density plots.</p>
<div class="section level4">
<h4 id="boxplot-mapped-reads">Boxplot Mapped Reads<a class="anchor" aria-label="anchor" href="#boxplot-mapped-reads"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/distribution_mapped_reads.html">distribution_mapped_reads</a></span><span class="op">(</span>
    <span class="va">object</span>, <span class="va">palette</span>, alpha <span class="op">=</span> <span class="fl">0.8</span>,
    type <span class="op">=</span> <span class="st">"boxplot"</span>
<span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
        legend.position <span class="op">=</span> <span class="st">"none"</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">40</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
    <span class="op">)</span>

<span class="va">plot</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/distribution_mapped_reads_boxplot-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="density-plot">Density plot<a class="anchor" aria-label="anchor" href="#density-plot"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/distribution_mapped_reads.html">distribution_mapped_reads</a></span><span class="op">(</span>
    <span class="va">object</span>, <span class="va">palette</span>, alpha <span class="op">=</span> <span class="fl">0.5</span>,
    type <span class="op">=</span> <span class="st">"density"</span>
<span class="op">)</span> <span class="op">+</span>
    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>

<span class="va">plot</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/distribution_mapped_reads_density-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="barcode-lost">Barcode Lost<a class="anchor" aria-label="anchor" href="#barcode-lost"></a>
</h4>
<p>Moreover another very important quality check when a Genetic
Screening is performed is to check the barcode lost during the
experiment. ScreenR implements a function able to compute and plot the
number of barcodes lost.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_barcode_lost.html">plot_barcode_lost</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span>, palette <span class="op">=</span> <span class="va">palette</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
        legend.position <span class="op">=</span> <span class="st">"none"</span>,
        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">40</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
    <span class="op">)</span>
<span class="va">plot</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_barcode_lost-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 class="tabset" id="plot-mds">Plot MDS<a class="anchor" aria-label="anchor" href="#plot-mds"></a>
</h4>
<p>In order to see compare the samples an initial MDS analysis can be
conducted. In ScreenR this can be done using the <em>plot_MDS</em>
function as follow.</p>
<div class="section level5">
<h5 id="for-sample">For Sample<a class="anchor" aria-label="anchor" href="#for-sample"></a>
</h5>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_MDS.html">plot_MDS</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span><span class="op">)</span> </code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Sample-1.png" width="960"></p>
</div>
<div class="section level5">
<h5 id="for-treatment">For Treatment<a class="anchor" aria-label="anchor" href="#for-treatment"></a>
</h5>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GGgroups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".*_"</span>, <span class="st">""</span>, <span class="va">groups</span><span class="op">)</span>

<span class="fu"><a href="../reference/plot_MDS.html">plot_MDS</a></span><span class="op">(</span>
    screenR_Object <span class="op">=</span> <span class="va">object</span>,
    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">GGgroups</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GGgroups</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Treatment-1.png" width="960"></p>
</div>
<div class="section level5">
<h5 id="for-day">For Day<a class="anchor" aria-label="anchor" href="#for-day"></a>
</h5>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GGgroups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"_.*"</span>, <span class="st">""</span>, <span class="va">groups</span><span class="op">)</span>
<span class="fu"><a href="../reference/plot_MDS.html">plot_MDS</a></span><span class="op">(</span>
    screenR_Object <span class="op">=</span> <span class="va">object</span>,
    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">GGgroups</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GGgroups</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Day-1.png" width="960"></p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="statistical-analysis">Statistical Analysis<a class="anchor" aria-label="anchor" href="#statistical-analysis"></a>
</h3>
<p>Finally can be conducted the real analysis. The statistical Analysis
is based on three methods:</p>
<ul>
<li>Z-score filtering</li>
<li>CAMERA filtering</li>
<li>ROAST filtering</li>
</ul>
<div class="section level4">
<h4 id="compute-metrics">Compute Metrics<a class="anchor" aria-label="anchor" href="#compute-metrics"></a>
</h4>
<p>In order to compute the Z-score, first a list of metrics has to be
computed. In particular a <em>Log2FC</em> is computed for the treated vs
control samples in the different conditions. Here for example a treated
vs control in different day is computed. Then the different distribution
of the Z-score can be plotted using the
<em>plot_Zscore_distribution</em> function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># 2DG</span>
<span class="va">data_with_measure_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/compute_metrics.html">compute_metrics</a></span><span class="op">(</span>
        <span class="va">object</span>, control <span class="op">=</span> <span class="st">"Time3"</span>, treatment <span class="op">=</span> <span class="st">"TRT"</span>,
        day <span class="op">=</span> <span class="st">"Time3"</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/compute_metrics.html">compute_metrics</a></span><span class="op">(</span>
        <span class="va">object</span>, control <span class="op">=</span> <span class="st">"Time4"</span>, treatment <span class="op">=</span> <span class="st">"TRT"</span>,
        day <span class="op">=</span> <span class="st">"Time4"</span>
    <span class="op">)</span>
<span class="op">)</span>


<span class="fu"><a href="../reference/plot_Zscore_distribution.html">plot_Zscore_distribution</a></span><span class="op">(</span><span class="va">data_with_measure_TRT</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/compute_metrics-1.png" width="960"></p>
</div>
<div class="section level4">
<h4 id="z-score-hit">Z-score hit<a class="anchor" aria-label="anchor" href="#z-score-hit"></a>
</h4>
<p>Now that the metrics has been computed the hits according to Z-score
can be found. This in ScreenR is done using the <em>find_zscore_hit</em>
function.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">zscore_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_zscore_hit.html">find_zscore_hit</a></span><span class="op">(</span>
        table_treate_vs_control <span class="op">=</span> <span class="va">data_with_measure_TRT</span><span class="op">$</span><span class="va">Time3</span>,
        number_barcode <span class="op">=</span> <span class="fl">7</span>, metric <span class="op">=</span> <span class="st">"median"</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_zscore_hit.html">find_zscore_hit</a></span><span class="op">(</span>
        table_treate_vs_control <span class="op">=</span> <span class="va">data_with_measure_TRT</span><span class="op">$</span><span class="va">Time4</span>,
        number_barcode <span class="op">=</span> <span class="fl">7</span>, metric <span class="op">=</span> <span class="st">"median"</span>
    <span class="op">)</span>
<span class="op">)</span>
<span class="va">zscore_hit_TRT</span></code></pre></div>
<pre><code><span class="co">## $Time3</span>
<span class="co">## <span style="color: #949494;"># A tibble: 30 × 2</span></span>
<span class="co">##    Gene     numberOfBarcode</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_116               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_120               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_128               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_156               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_173               9</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_190               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_193               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_200               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_218               8</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_226               8</span>
<span class="co">## <span style="color: #949494;"># … with 20 more rows</span></span>
<span class="co">## </span>
<span class="co">## $Time4</span>
<span class="co">## <span style="color: #949494;"># A tibble: 30 × 2</span></span>
<span class="co">##    Gene     numberOfBarcode</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_121               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_147               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_148               9</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_15                9</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_168               9</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_171               9</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_173               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_185               8</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_204               8</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_245              18</span>
<span class="co">## <span style="color: #949494;"># … with 20 more rows</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="camera">CAMERA<a class="anchor" aria-label="anchor" href="#camera"></a>
</h4>
<p>The same can be done with the CAMERA hit using the function
<em>find_camera_hit</em>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">matrix_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fl">0</span> <span class="op">+</span> <span class="va">groups</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">matrix_model</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span>

<span class="va">camera_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_camera_hit.html">find_camera_hit</a></span><span class="op">(</span>
        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,
        contrast <span class="op">=</span> <span class="st">"Time3_TRT"</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_camera_hit.html">find_camera_hit</a></span><span class="op">(</span>
        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,
        contrast <span class="op">=</span> <span class="st">"Time4_TRT"</span>
    <span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in .zscoreDGE(y = y, design = design, contrast = contrast): 3rows with</span>
<span class="co">## all zero counts</span></code></pre>
<pre><code><span class="co">## Warning: Zero sample variances detected, have been offset away from zero</span></code></pre>
<pre><code><span class="co">## Warning in .zscoreDGE(y = y, design = design, contrast = contrast): 3rows with</span>
<span class="co">## all zero counts</span></code></pre>
<pre><code><span class="co">## Warning: Zero sample variances detected, have been offset away from zero</span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">camera_hit_TRT</span></code></pre></div>
<pre><code><span class="co">## $Time3</span>
<span class="co">## <span style="color: #949494;"># A tibble: 281 × 5</span></span>
<span class="co">##    Gene     NGenes Direction PValue   FDR</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_199     10 Down      0.013<span style="text-decoration: underline;">1</span> 0.522</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_239     10 Down      0.018<span style="text-decoration: underline;">9</span> 0.674</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_78      10 Down      0.032<span style="text-decoration: underline;">4</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_478     10 Down      0.041<span style="text-decoration: underline;">4</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_508      9 Down      0.044<span style="text-decoration: underline;">7</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_516      9 Down      0.052<span style="text-decoration: underline;">8</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_298     10 Down      0.058<span style="text-decoration: underline;">4</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_347     10 Down      0.059<span style="text-decoration: underline;">4</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_183     10 Down      0.064<span style="text-decoration: underline;">6</span> 0.906</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_245     32 Down      0.069<span style="text-decoration: underline;">8</span> 0.906</span>
<span class="co">## <span style="color: #949494;"># … with 271 more rows</span></span>
<span class="co">## </span>
<span class="co">## $Time4</span>
<span class="co">## <span style="color: #949494;"># A tibble: 269 × 5</span></span>
<span class="co">##    Gene     NGenes Direction PValue   FDR</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_199     10 Down      0.013<span style="text-decoration: underline;">1</span> 0.779</span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_239     10 Down      0.020<span style="text-decoration: underline;">1</span> 0.941</span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_78      10 Down      0.028<span style="text-decoration: underline;">5</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_298     10 Down      0.031<span style="text-decoration: underline;">6</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_508      9 Down      0.035<span style="text-decoration: underline;">5</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_516      9 Down      0.037<span style="text-decoration: underline;">0</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_494      9 Down      0.043<span style="text-decoration: underline;">7</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_478     10 Down      0.049<span style="text-decoration: underline;">1</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_257     10 Down      0.050<span style="text-decoration: underline;">0</span> 0.969</span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_183     10 Down      0.058<span style="text-decoration: underline;">4</span> 0.969</span>
<span class="co">## <span style="color: #949494;"># … with 259 more rows</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="roast">ROAST<a class="anchor" aria-label="anchor" href="#roast"></a>
</h4>
<p>Last but not least this is done also for the ROAST hit using the
function<br><em>find_roast_hit</em>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roast_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_roast_hit.html">find_roast_hit</a></span><span class="op">(</span>
        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,
        contrast <span class="op">=</span> <span class="st">"Time3_TRT"</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_roast_hit.html">find_roast_hit</a></span><span class="op">(</span>
        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,
        contrast <span class="op">=</span> <span class="st">"Time4_TRT"</span>
    <span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in .zscoreDGE(y = y, design = design, contrast = contrast): 3rows with</span>
<span class="co">## all zero counts</span>

<span class="co">## Warning in .zscoreDGE(y = y, design = design, contrast = contrast): 3rows with</span>
<span class="co">## all zero counts</span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">roast_hit_TRT</span></code></pre></div>
<pre><code><span class="co">## $Time3</span>
<span class="co">## <span style="color: #949494;"># A tibble: 534 × 9</span></span>
<span class="co">##    Gene     NGenes PropDown PropUp Direction PValue   FDR PValue.Mixed FDR.Mixed</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_245     32        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_364     12        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_372     12        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_380     10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_381     10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_1       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_2       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_3       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_4       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_5       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #949494;"># … with 524 more rows</span></span>
<span class="co">## </span>
<span class="co">## $Time4</span>
<span class="co">## <span style="color: #949494;"># A tibble: 534 × 9</span></span>
<span class="co">##    Gene     NGenes PropDown PropUp Direction PValue   FDR PValue.Mixed FDR.Mixed</span>
<span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 1</span> Gene_245     32        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 2</span> Gene_364     12        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 3</span> Gene_372     12        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 4</span> Gene_380     10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 5</span> Gene_381     10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 6</span> Gene_1       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 7</span> Gene_2       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 8</span> Gene_3       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;"> 9</span> Gene_4       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #BCBCBC;">10</span> Gene_5       10        1      0 Down      0.000<span style="text-decoration: underline;">1</span>  1<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0.000<span style="text-decoration: underline;">1</span>    0.000<span style="text-decoration: underline;">1</span></span>
<span class="co">## <span style="color: #949494;"># … with 524 more rows</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="find-common-hit">Find Common Hit<a class="anchor" aria-label="anchor" href="#find-common-hit"></a>
</h4>
<p>ScreenR consider as final hit only the one present in common among
the three methods. In particular two strategy are possible:</p>
<ul>
<li>Are considered candidate Hits the one present in at least two of the
three methods</li>
<li>Are considered candidate Hits the one present in all of the three
methods</li>
</ul>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">common_hit_TRT_at_least_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span>
        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Day3</span>,
        common_in <span class="op">=</span> <span class="fl">2</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span>
        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Day6</span>,
        common_in <span class="op">=</span> <span class="fl">2</span>
    <span class="op">)</span>
<span class="op">)</span>

<span class="va">common_hit_TRT_at_least_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span>
        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>,
        common_in <span class="op">=</span> <span class="fl">3</span>
    <span class="op">)</span>,
    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span>
        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>,
        common_in <span class="op">=</span> <span class="fl">3</span>
    <span class="op">)</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plot-common-hit">Plot common hit<a class="anchor" aria-label="anchor" href="#plot-common-hit"></a>
</h4>
<p>The hits can be easily visualized using the <em>plot_common_hit</em>
function.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_common_hit.html">plot_common_hit</a></span><span class="op">(</span>
    hit_zscore <span class="op">=</span> <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, hit_camera <span class="op">=</span> <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>,
    <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>
<span class="op">)</span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Venn_diagram_in_at_least_2-1.png" width="960"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.2.0 RC (2022-04-14 r82180)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur/Monterey 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">## [1] ScreenR_0.99.41  tidyr_1.2.0      dplyr_1.0.8      ggplot2_3.3.5   </span>
<span class="co">## [5] BiocStyle_2.23.1</span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] tidyselect_1.1.2    locfit_1.5-9.5      xfun_0.30          </span>
<span class="co">##  [4] bslib_0.3.1         purrr_0.3.4         splines_4.2.0      </span>
<span class="co">##  [7] lattice_0.20-45     colorspace_2.0-3    vctrs_0.4.1        </span>
<span class="co">## [10] generics_0.1.2      htmltools_0.5.2     yaml_2.3.5         </span>
<span class="co">## [13] utf8_1.2.2          rlang_1.0.2         pkgdown_2.0.2      </span>
<span class="co">## [16] jquerylib_0.1.4     pillar_1.7.0        glue_1.6.2         </span>
<span class="co">## [19] withr_2.5.0         lifecycle_1.0.1     stringr_1.4.0      </span>
<span class="co">## [22] munsell_0.5.0       gtable_0.3.0        ragg_1.2.2         </span>
<span class="co">## [25] memoise_2.0.1       evaluate_0.15       labeling_0.4.2     </span>
<span class="co">## [28] knitr_1.38          fastmap_1.1.0       fansi_1.0.3        </span>
<span class="co">## [31] highr_0.9           Rcpp_1.0.8.3        edgeR_3.37.1       </span>
<span class="co">## [34] scales_1.2.0        BiocManager_1.30.16 cachem_1.0.6       </span>
<span class="co">## [37] limma_3.51.8        desc_1.4.1          jsonlite_1.8.0     </span>
<span class="co">## [40] farver_2.1.0        systemfonts_1.0.4   fs_1.5.2           </span>
<span class="co">## [43] textshaping_0.3.6   digest_0.6.29       stringi_1.7.6      </span>
<span class="co">## [46] bookdown_0.26       grid_4.2.0          rprojroot_2.0.3    </span>
<span class="co">## [49] ggvenn_0.1.9        cli_3.2.0           tools_4.2.0        </span>
<span class="co">## [52] magrittr_2.0.3      sass_0.4.1          tibble_3.1.6       </span>
<span class="co">## [55] crayon_1.5.1        pkgconfig_2.0.3     ellipsis_0.3.2     </span>
<span class="co">## [58] rmarkdown_2.13      rstudioapi_0.13     R6_2.5.1           </span>
<span class="co">## [61] compiler_4.2.0</span></code></pre>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Emanuel Michele Soda.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
