<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="ScreenR">
<title>ScreenR Example Analysis • ScreenR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ScreenR Example Analysis">
<meta property="og:description" content="ScreenR">
<meta property="og:image" content="https://emanuelsoda.github.io/ScreenR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">ScreenR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.54</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Analysis_Example.html">ScreenR Example Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/EmanuelSoda/ScreenR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>ScreenR Example Analysis</h1>
                        <h4 data-toc-skip class="author">Emanuel Michele
Soda</h4>
            <address class="author_afil">
      Istituto Europeo Oncologia (IEO), Milano,
Italy<br><a class="author_email" href="mailto:#"></a><a href="mailto:emanuelmichele@ieo.it" class="email">emanuelmichele@ieo.it</a>
      </address>
                              <h4 data-toc-skip class="author">Elena
Ceccacci</h4>
            <address class="author_afil">
      Istituto Europeo Oncologia (IEO), Milano,
Italy<br><a class="author_email" href="mailto:#"></a><a href="mailto:elena.ceccacci@ieo.it" class="email">elena.ceccacci@ieo.it</a>
      </address>
                  
            <h4 data-toc-skip class="date">2023-09-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/EmanuelSoda/ScreenR/blob/HEAD/vignettes/Analysis_Example.Rmd" class="external-link"><code>vignettes/Analysis_Example.Rmd</code></a></small>
      <div class="d-none name"><code>Analysis_Example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="importing-package">Importing Package<a class="anchor" aria-label="anchor" href="#importing-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>collapse <span class="op">=</span> <span class="cn">TRUE</span>, comment <span class="op">=</span> <span class="st">"#&gt;"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The R package <strong>ScreenR</strong> has been developed to perform
analysis of High-throughput RNA interference screening using pooled
shRNA libraries and next generation sequencing.. Nowadays several short
hairpin RNA (shRNA) libraries are commercial available, and in the last
years the interest in this type of analysis, often called barcode
screening, has greatly increased for their benefits both from a
time-consuming point of view and for the possibility of carrying out
screening on a large number of genes simultaneously. However, the
bioinformatic analysis of this type of screening still lacks a golden
standard. Here, ScreenR allows the user to carry out a preliminary
quality check of the experiment, visually inspect the data and finally
identify the most significant hits of the experiment through a series of
plots and cross-statistical analyses.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p><strong>ScreenR</strong> requires several CRAN and Bioconductor R
packages to be installed. Dependencies are usually handled
automatically, when installing the package using the following
commands:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ScreenR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="github">Github<a class="anchor" aria-label="anchor" href="#github"></a>
</h3>
<p>The newest version can directly be installed from GitHub using the
CRAN package devtools:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"EmanuelSoda/ScreenR"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<p>Here is reported the ScreenR pipeline
<img src="../../../../../../Users/ieo5571/Documents/IEO/Stage/ScreenR/reference/figures/Pipeline.png" align="top"></p>
<div class="section level3">
<h3 id="loading-the-package">Loading the package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<p>After installation, loading the package is simple as :</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://emanuelsoda.github.io/ScreenR/">ScreenR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Welcome to ScreenR</span></span>
<span><span class="co">#&gt; --------------------------</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; See ?ScreenR for a list of</span></span>
<span><span class="co">#&gt; available functions.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Enjoy!</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="read-data">Read Data<a class="anchor" aria-label="anchor" href="#read-data"></a>
</h3>
<p>The input of ScreenR is a count table obtained from barcode alignment
to the reference genome/library. A count table is usually the starting
point of an RNA-seq deferentially expressed genes analysis and consists
of a matrix containing reads count organized with:</p>
<ul>
<li>Genes on the rows</li>
<li>Samples on the columns</li>
</ul>
<p>For this vignette we will use as an example a Loss of Function
Chemical lethality Genetic Screening performed using shRNA libraries
where each gene is represented by ten slightly different shRNA each
labeled with a unique barcode coming from an unpublished dataset
generated using the <a href="https://cellecta.com/" class="external-link">Cellecta</a>
protocol.</p>
<p>First of all the data has to be read. Then another very important
step is to set up the column names in the following way:</p>
<ul>
<li>The first part of the string is the timepoint</li>
<li>The second is the type of sample (example: treated or control or the
name of the treatment)</li>
<li>Third slot is a way to make each replicate unique (for example a
letter or a number)</li>
</ul>
<p>In the end we have</p>
<p><strong><code>Time point_Type of sample_replicate</code></strong></p>
<p>and for example we could have <strong>h24_control_1</strong> which
mean the first replicate of the control sample at 24 hour.</p>
<p>Since this dataset comes from a Chemical Synthetic Lethality
experiments the samples treated with the drug combined with the shRNAs
knockdown should present a decreased number of reads compared to the
controls.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">count_table</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">annotation_table</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">count_table</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Barcode"</span>, <span class="st">"Time1"</span>, <span class="st">"Time2"</span>, <span class="st">"Time3_TRT_A"</span>, <span class="st">"Time3_TRT_B"</span>, <span class="st">"Time3_TRT_C"</span>,</span>
<span>    <span class="st">"Time3_CTRL_A"</span>, <span class="st">"Time3_CTRL_B"</span>, <span class="st">"Time3_CTRL_C"</span>, </span>
<span>    <span class="st">"Time4_TRT_A"</span>, <span class="st">"Time4_TRT_B"</span>, <span class="st">"Time4_TRT_C"</span>, </span>
<span>    <span class="st">"Time4_CTRL_A"</span>, <span class="st">"Time4_CTRL_B"</span>, <span class="st">"Time4_CTRL_C"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">Barcode</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Barcode</span> <span class="op">!=</span> <span class="st">"*"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  </span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">total_Annotation</span> <span class="op">&lt;-</span> <span class="va">annotation_table</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="object-creation">Object Creation<a class="anchor" aria-label="anchor" href="#object-creation"></a>
</h3>
<p>The second needed step is to create a <strong>ScreenR object</strong>
from the count table. The ScreenR object is created using the function
<strong>create_screenr_object()</strong> and will be used to store the
most important information to perform the analysis. Most of the ScreenR
function takes as main input the ScreenR object to perform the needed
operation and return a result.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"T1/T2"</span>, <span class="st">"T1/T2"</span>,</span>
<span>    <span class="st">"Time3_TRT"</span>, <span class="st">"Time3_TRT"</span>, <span class="st">"Time3_TRT"</span>,</span>
<span>    <span class="st">"Time3_CTRL"</span>, <span class="st">"Time3_CTRL"</span>, <span class="st">"Time3_CTRL"</span>,</span>
<span>    <span class="st">"Time4_TRT"</span>, <span class="st">"Time4_TRT"</span>, <span class="st">"Time4_TRT"</span>,</span>
<span>    <span class="st">"Time4_CTRL"</span>, <span class="st">"Time4_CTRL"</span>, <span class="st">"Time4_CTRL"</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#66c2a5"</span>, <span class="st">"#fc8d62"</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#8da0cb"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#e78ac3"</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#a6d854"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#ffd92f"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_screenr_object.html">create_screenr_object</a></span><span class="op">(</span></span>
<span>    table <span class="op">=</span> <span class="va">data</span>, annotation <span class="op">=</span> <span class="va">total_Annotation</span>, groups <span class="op">=</span> <span class="va">groups</span>,</span>
<span>    replicates <span class="op">=</span> <span class="st">""</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="removing-all-zero-rows">Removing all zero rows<a class="anchor" aria-label="anchor" href="#removing-all-zero-rows"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_all_zero_row.html">remove_all_zero_row</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="computing-the-needed-tables">Computing the needed tables<a class="anchor" aria-label="anchor" href="#computing-the-needed-tables"></a>
</h3>
<p>Once the object is created, the data must be normalized to start the
analysis. ScreenR uses <em>Counts Per Million</em>
(<strong>CPM</strong>) normalization which has the following
mathematical expression:</p>
<p><span class="math display">\[CPM = \frac{Number \; of \; mapped \;
reads \; to \; a \; barcode}
             { \sum_{sample}{Number\; of \;mapped \; reads}}
*10^{6}\]</span></p>
<p>The number of reads mapped for each barcode in a sample are
normalized by the number of reads in that sample and multiplied by one
million.</p>
<p>This information is store in a <em>data table</em> which is a tidy
version of the original <em>count table</em> and will be used throughout
the analysis.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_data.html">normalize_data</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_data_table.html">compute_data_table</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quality-check">Quality Check<a class="anchor" aria-label="anchor" href="#quality-check"></a>
</h3>
<p>The first step to perform when dealing with sequencing data is to
check the quality of the samples. In ScreenR this can be done using
several methods.</p>
<div class="section level4">
<h4 id="mapped-reads">Mapped Reads<a class="anchor" aria-label="anchor" href="#mapped-reads"></a>
</h4>
<p>The total number of mapped reads can be displayed with a barplot with
the formula.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_mapped_reads.html">plot_mapped_reads</a></span><span class="op">(</span><span class="va">object</span>, <span class="va">palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/comma.html" class="external-link">comma</a></span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Number of Mapped Reads in each sample"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plot</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_mapped_reads-1.png" width="700"></p>
<p>For example the distribution can be seen using both boxplots or
density plots.</p>
<p>#### Boxplot Mapped Reads</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_mapped_reads_distribution.html">plot_mapped_reads_distribution</a></span><span class="op">(</span></span>
<span>    <span class="va">object</span>, <span class="va">palette</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"boxplot"</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">plot</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_mapped_reads_distribution_boxplot-1.png" width="700"></p>
<div class="section level5">
<h5 id="density-plot">Density plot<a class="anchor" aria-label="anchor" href="#density-plot"></a>
</h5>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_mapped_reads_distribution.html">plot_mapped_reads_distribution</a></span><span class="op">(</span></span>
<span>    <span class="va">object</span>, <span class="va">palette</span>,</span>
<span>    alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"density"</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">plot</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_mapped_reads_distribution_density-1.png" width="700"></p>
</div>
</div>
<div class="section level4">
<h4 id="barcode-lost">Barcode Lost<a class="anchor" aria-label="anchor" href="#barcode-lost"></a>
</h4>
<p>Another very important quality check when a Genetic Screening is
performed is to check the barcode lost during the experiment, meaning
the barcode that after different time points or treatments results in
reads count equal to zero. ScreenR implements a function able to compute
and plot the number of barcodes lost for each samples.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_barcode_lost.html">plot_barcode_lost</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span>, palette <span class="op">=</span> <span class="va">palette</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_barcode_lost-1.png" width="700"></p>
<p>Moreover it is important to check if the lost barcodes in a sample
all belong to the same gene, in order to verify that an adequate number
of barcodes per gene are still present. This can be done by visualizing
the number of barcode lost in a sample by gene.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_barcode_lost_for_gene.html">plot_barcode_lost_for_gene</a></span><span class="op">(</span><span class="va">object</span>,</span>
<span>    samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time4_TRT_C"</span>, <span class="st">"Time4_CTRL_C"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">plot</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_barcode_lost_for_gene-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 class="tabset" id="plot-mds">Plot MDS<a class="anchor" aria-label="anchor" href="#plot-mds"></a>
</h4>
<p>In order to see the samples clusterization an initial MDS analysis
can be conducted. In ScreenR this can be done using the
<em>plot_mds</em> function and the user can decide the color code of the
graph in order to highlight the trend of the samples based on
replicates, treatment or timepoints simply by modifying the field levels
in the plot_mds function.</p>
<div class="section level5">
<h5 id="for-sample">For Sample<a class="anchor" aria-label="anchor" href="#for-sample"></a>
</h5>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_mds.html">plot_mds</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Sample-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="for-treatment">For Treatment<a class="anchor" aria-label="anchor" href="#for-treatment"></a>
</h5>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_data_table.html">get_data_table</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span>   <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">Day</span>, <span class="va">Treatment</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ScreenR normalized data table containing:</span></span>
<span><span class="co">#&gt;  74438 rows</span></span>
<span><span class="co">#&gt;  9 columns</span></span>
<span></span>
<span><span class="va">group_treatment</span> <span class="op">&lt;-</span> <span class="va">group_table</span><span class="op">$</span><span class="va">Treatment</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_mds.html">plot_mds</a></span><span class="op">(</span></span>
<span>    screenR_Object <span class="op">=</span> <span class="va">object</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">group_treatment</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">group_treatment</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Treatment-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="for-day">For Day<a class="anchor" aria-label="anchor" href="#for-day"></a>
</h5>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group_day</span> <span class="op">&lt;-</span> <span class="va">group_table</span><span class="op">$</span><span class="va">Day</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_mds.html">plot_mds</a></span><span class="op">(</span></span>
<span>    screenR_Object <span class="op">=</span> <span class="va">object</span>,</span>
<span>    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">group_day</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">group_day</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Plot_MDS_Day-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="statistical-analysis">Statistical Analysis<a class="anchor" aria-label="anchor" href="#statistical-analysis"></a>
</h3>
<p>Once the various steps of the quality check have been passed, the
actual statistical analysis can begin. The statistical Analysis of
ScreenR is based on three different methods:</p>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/21515799/" class="external-link">Z-score
filtering</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3458527/" class="external-link">CAMERA
filtering</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2922896/" class="external-link">ROAST
filtering</a></li>
</ul>
<div class="section level4">
<h4 id="z-score-hit">Z-score hit<a class="anchor" aria-label="anchor" href="#z-score-hit"></a>
</h4>
<p>In order to compute the Z-score, first a list of metrics has to be
computed. In particular a <em>Log2FC</em> is computed for the treated vs
control samples in the different conditions. This can be done using the
function <em>compute_metrics()</em>. Here is reported an example of
treated vs control in different day.</p>
<p>Then the different distribution of the Z-score can be plotted using
the <em>plot_zscore_distribution</em> function.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 2DG</span></span>
<span><span class="va">data_with_measure_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/compute_metrics.html">compute_metrics</a></span><span class="op">(</span></span>
<span>        <span class="va">object</span>,</span>
<span>        control <span class="op">=</span> <span class="st">"CTRL"</span>, treatment <span class="op">=</span> <span class="st">"TRT"</span>,</span>
<span>        day <span class="op">=</span> <span class="st">"Time3"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/compute_metrics.html">compute_metrics</a></span><span class="op">(</span></span>
<span>        <span class="va">object</span>,</span>
<span>        control <span class="op">=</span> <span class="st">"CTRL"</span>, treatment <span class="op">=</span> <span class="st">"TRT"</span>,</span>
<span>        day <span class="op">=</span> <span class="st">"Time4"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_zscore_distribution.html">plot_zscore_distribution</a></span><span class="op">(</span><span class="va">data_with_measure_TRT</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span> </span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/compute_metrics-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="z-score-hit-1">Z-score hit<a class="anchor" aria-label="anchor" href="#z-score-hit-1"></a>
</h4>
<p>Based on these metrics the Z-score hit identification can be computed
using the <em>find_zscore_hit</em> function.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">zscore_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_zscore_hit.html">find_zscore_hit</a></span><span class="op">(</span></span>
<span>        table_treate_vs_control <span class="op">=</span> <span class="va">data_with_measure_TRT</span><span class="op">$</span><span class="va">Time3</span>,</span>
<span>        number_barcode <span class="op">=</span> <span class="fl">6</span>, metric <span class="op">=</span> <span class="st">"median"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_zscore_hit.html">find_zscore_hit</a></span><span class="op">(</span></span>
<span>        table_treate_vs_control <span class="op">=</span> <span class="va">data_with_measure_TRT</span><span class="op">$</span><span class="va">Time4</span>,</span>
<span>        number_barcode <span class="op">=</span> <span class="fl">6</span>, metric <span class="op">=</span> <span class="st">"median"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">zscore_hit_TRT</span></span>
<span><span class="co">#&gt; $Time3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 83 × 2</span></span></span>
<span><span class="co">#&gt;    Gene     numberOfBarcode</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_1                 7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_11                7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_116               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_120               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_121               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_128               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_15                7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_156               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_158               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_160               7</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 73 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Time4</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 97 × 2</span></span></span>
<span><span class="co">#&gt;    Gene     numberOfBarcode</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_107               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_121               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_125               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_126               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_134               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_138               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_139               7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_14                7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_147               8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_148               9</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 87 more rows</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="camera">CAMERA<a class="anchor" aria-label="anchor" href="#camera"></a>
</h4>
<p>The same can be done with the CAMERA hit using the function
<em>find_camera_hit</em>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">matrix_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">matrix_model</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">camera_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_camera_hit.html">find_camera_hit</a></span><span class="op">(</span></span>
<span>        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,</span>
<span>        contrast <span class="op">=</span> <span class="st">"Time3_TRT"</span>,</span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_camera_hit.html">find_camera_hit</a></span><span class="op">(</span></span>
<span>        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,</span>
<span>        contrast <span class="op">=</span> <span class="st">"Time4_TRT"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">camera_hit_TRT</span></span>
<span><span class="co">#&gt; $Time3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 258 × 5</span></span></span>
<span><span class="co">#&gt;    Gene     NGenes Direction   PValue      FDR</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_372     12 Down      3.80<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span> 1.93<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_364     12 Down      7.23<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span> 1.93<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_374     10 Down      3.55<span style="color: #949494;">e</span><span style="color: #BB0000;">- 7</span> 6.32<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_282     10 Down      2.24<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span> 2.40<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_75      10 Down      7.59<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span> 6.76<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_323     10 Down      1.45<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> 1.11<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_179     10 Down      1.68<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_117     10 Down      1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 4.23<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_228     10 Down      1.49<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 4.97<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_211     10 Down      1.84<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 5.77<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 248 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Time4</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 241 × 5</span></span></span>
<span><span class="co">#&gt;    Gene     NGenes Direction   PValue      FDR</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_364     12 Down      1.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-24</span> 6.03<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_372     12 Down      3.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 9.24<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_374     10 Down      9.55<span style="color: #949494;">e</span><span style="color: #BB0000;">- 8</span> 1.70<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_323     10 Down      7.27<span style="color: #949494;">e</span><span style="color: #BB0000;">- 7</span> 9.70<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_282     10 Down      6.22<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span> 6.64<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_75      10 Down      1.76<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span> 1.57<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_296     10 Down      1.31<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span> 1.00<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_80      10 Down      5.72<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span> 3.82<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_54      10 Down      7.94<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> 3.26<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_179     10 Down      1.13<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 3.94<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 231 more rows</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="roast">ROAST<a class="anchor" aria-label="anchor" href="#roast"></a>
</h4>
<p>Last but not least this is done also for the ROAST hit using the
function<br><em>find_roast_hit</em>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">roast_hit_TRT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_roast_hit.html">find_roast_hit</a></span><span class="op">(</span></span>
<span>        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,</span>
<span>        contrast <span class="op">=</span> <span class="st">"Time3_TRT"</span>, </span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_roast_hit.html">find_roast_hit</a></span><span class="op">(</span></span>
<span>        screenR_Object <span class="op">=</span> <span class="va">object</span>, matrix_model <span class="op">=</span> <span class="va">matrix_model</span>,</span>
<span>        contrast <span class="op">=</span> <span class="st">"Time4_TRT"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">roast_hit_TRT</span></span>
<span><span class="co">#&gt; $Time3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 30 × 9</span></span></span>
<span><span class="co">#&gt;    Gene    NGenes PropDown PropUp Direction PValue    FDR PValue.Mixed FDR.Mixed</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_3…     12    0.917    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.013<span style="text-decoration: underline;">4</span>       0.000<span style="text-decoration: underline;">1</span>    0.013<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_3…     12    0.833    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.013<span style="text-decoration: underline;">4</span>       0.000<span style="text-decoration: underline;">1</span>    0.013<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_3…     10    0.7      0   Down      0.000<span style="text-decoration: underline;">2</span> 0.020<span style="text-decoration: underline;">0</span>       0.000<span style="text-decoration: underline;">8</span>    0.083<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_88     10    0.5      0.1 Down      0.000<span style="text-decoration: underline;">3</span> 0.026<span style="text-decoration: underline;">7</span>       0.002<span style="text-decoration: underline;">5</span>    0.145 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_1…     10    0.3      0   Down      0.000<span style="text-decoration: underline;">6</span> 0.048<span style="text-decoration: underline;">9</span>       0.016<span style="text-decoration: underline;">7</span>    0.222 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_75     10    0.4      0   Down      0.000<span style="text-decoration: underline;">8</span> 0.050<span style="text-decoration: underline;">1</span>       0.021<span style="text-decoration: underline;">1</span>    0.244 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_2…     10    0.5      0   Down      0.001<span style="text-decoration: underline;">2</span> 0.068<span style="text-decoration: underline;">2</span>       0.026<span style="text-decoration: underline;">3</span>    0.264 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_2…     10    0.2      0   Down      0.004<span style="text-decoration: underline;">6</span> 0.162        0.075<span style="text-decoration: underline;">2</span>    0.410 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_4…     10    0.4      0.1 Down      0.006<span style="text-decoration: underline;">6</span> 0.206        0.010<span style="text-decoration: underline;">8</span>    0.162 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_3…     10    0.2      0   Down      0.008<span style="text-decoration: underline;">5</span> 0.222        0.019<span style="text-decoration: underline;">3</span>    0.232 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 20 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Time4</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 105 × 9</span></span></span>
<span><span class="co">#&gt;    Gene   NGenes PropDown PropUp Direction PValue     FDR PValue.Mixed FDR.Mixed</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Gene_…     12      1      0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Gene_…     12      1      0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Gene_…     10      0.9    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene_…     10      0.9    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Gene_…     10      0.7    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Gene_…     10      0.7    0.1 Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Gene_…     10      0.7    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">1</span>  0.000<span style="text-decoration: underline;">785</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Gene_…     10      0.6    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">2</span>  0.001<span style="text-decoration: underline;">57</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Gene_…     10      0.4    0.1 Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.000<span style="text-decoration: underline;">5</span>  0.002<span style="text-decoration: underline;">93</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Gene_…     10      0.4    0   Down      0.000<span style="text-decoration: underline;">1</span> 0.002<span style="text-decoration: underline;">05</span>       0.013<span style="text-decoration: underline;">5</span>  0.027<span style="text-decoration: underline;">4</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 95 more rows</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="find-common-hit">Find Common Hit<a class="anchor" aria-label="anchor" href="#find-common-hit"></a>
</h4>
<p>ScreenR considers as final hit only the one result as candidate hit
in all three statistical methods. However this is a particularly
stringent method and in some cases leads to a small number of results.
For this reason the user can also decide to opt for a less stringent
method that considers only the hits present in at least two of the
statistical methods. The two different strategies can be computed with
the function::</p>
<ul>
<li><p><strong>common_hit_TRT_at_least_2</strong>: considering candidate
Hits the one present in at least two of the three methods (less
stringent)</p></li>
<li><p><strong>common_hit_TRT_at_least_3</strong>: considering candidate
Hits the one present in all of the three methods</p></li>
</ul>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">common_hit_TRT_at_least_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span></span>
<span>        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Day3</span>,</span>
<span>        common_in <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span></span>
<span>        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Day6</span>,</span>
<span>        common_in <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">common_hit_TRT_at_least_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    Time3 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span></span>
<span>        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time3</span>,</span>
<span>        common_in <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span>,</span>
<span>    Time4 <span class="op">=</span> <span class="fu"><a href="../reference/find_common_hit.html">find_common_hit</a></span><span class="op">(</span></span>
<span>        <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>,</span>
<span>        common_in <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plot-common-hit">Plot common hit<a class="anchor" aria-label="anchor" href="#plot-common-hit"></a>
</h4>
<p>The intersection of the hits found by the three statistical methods
can be easily visualized using the <em>plot_common_hit</em>
function.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_common_hit.html">plot_common_hit</a></span><span class="op">(</span></span>
<span>    hit_zscore <span class="op">=</span> <span class="va">zscore_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, hit_camera <span class="op">=</span> <span class="va">camera_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>,</span>
<span>    <span class="va">roast_hit_TRT</span><span class="op">$</span><span class="va">Time4</span>, show_elements <span class="op">=</span> <span class="cn">FALSE</span>, show_percentage <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/Venn_diagram_in_at_least_2-1.png" width="700"></p>
<p>As we all know, when we deal with statistical methods the is the
possibility of <em>type I</em> error also known as “false positive”. For
this reason is important to visualize the results obtained. This can be
done by visualizing the trend of the candidate hits obtained using the
function <strong>plot_trend</strong>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">candidate_hits</span> <span class="op">&lt;-</span> <span class="va">common_hit_TRT_at_least_2</span><span class="op">$</span><span class="va">Time3</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_trend.html">plot_trend</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span>, </span>
<span>           genes <span class="op">=</span> <span class="va">candidate_hits</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>           nrow <span class="op">=</span> <span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_var <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time1"</span>, <span class="st">"Time2"</span>, <span class="st">"TRT"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_trend-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_trend.html">plot_trend</a></span><span class="op">(</span>screenR_Object <span class="op">=</span> <span class="va">object</span>, </span>
<span>           genes <span class="op">=</span> <span class="va">candidate_hits</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>           nrow <span class="op">=</span> <span class="fl">2</span>, ncol <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           group_var <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Time1"</span>, <span class="st">"Time2"</span>, <span class="st">"TRT"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Analysis_Example_files/figure-html/plot_trend-2.png" width="700"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ScreenR_0.99.54  tidyr_1.3.0      dplyr_1.1.3      ggplot2_3.4.3   </span></span>
<span><span class="co">#&gt; [5] BiocStyle_2.28.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.7          utf8_1.2.3          generics_0.1.3     </span></span>
<span><span class="co">#&gt;  [4] stringi_1.7.12      lattice_0.21-8      digest_0.6.33      </span></span>
<span><span class="co">#&gt;  [7] magrittr_2.0.3      evaluate_0.21       grid_4.3.1         </span></span>
<span><span class="co">#&gt; [10] bookdown_0.35       fastmap_1.1.1       Matrix_1.5-4.1     </span></span>
<span><span class="co">#&gt; [13] rprojroot_2.0.3     jsonlite_1.8.7      limma_3.56.2       </span></span>
<span><span class="co">#&gt; [16] BiocManager_1.30.22 mgcv_1.8-42         purrr_1.0.2        </span></span>
<span><span class="co">#&gt; [19] fansi_1.0.4         scales_1.2.1        textshaping_0.3.6  </span></span>
<span><span class="co">#&gt; [22] jquerylib_0.1.4     cli_3.6.1           rlang_1.1.1        </span></span>
<span><span class="co">#&gt; [25] splines_4.3.1       munsell_0.5.0       withr_2.5.0        </span></span>
<span><span class="co">#&gt; [28] cachem_1.0.8        yaml_2.3.7          tools_4.3.1        </span></span>
<span><span class="co">#&gt; [31] memoise_2.0.1       colorspace_2.1-0    locfit_1.5-9.8     </span></span>
<span><span class="co">#&gt; [34] ggvenn_0.1.10       vctrs_0.6.3         R6_2.5.1           </span></span>
<span><span class="co">#&gt; [37] lifecycle_1.0.3     stringr_1.5.0       edgeR_3.42.4       </span></span>
<span><span class="co">#&gt; [40] fs_1.6.3            ragg_1.2.5          pkgconfig_2.0.3    </span></span>
<span><span class="co">#&gt; [43] desc_1.4.2          pkgdown_2.0.7       pillar_1.9.0       </span></span>
<span><span class="co">#&gt; [46] bslib_0.5.1         gtable_0.3.4        glue_1.6.2         </span></span>
<span><span class="co">#&gt; [49] Rcpp_1.0.11         systemfonts_1.0.4   highr_0.10         </span></span>
<span><span class="co">#&gt; [52] xfun_0.40           tibble_3.2.1        tidyselect_1.2.0   </span></span>
<span><span class="co">#&gt; [55] knitr_1.43          farver_2.1.1        nlme_3.1-162       </span></span>
<span><span class="co">#&gt; [58] patchwork_1.1.3     htmltools_0.5.6     labeling_0.4.3     </span></span>
<span><span class="co">#&gt; [61] rmarkdown_2.24      compiler_4.3.1</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Emanuel Michele Soda, Elena Ceccacci, Saverio Minucci.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
